synapse/tools/layer/dump.py

```diff
@@ -0,0 +1,186 @@
+
+        # Handle no edits to export
+        if soffs >= await layer.getEditIndx():
+            mesg = f'No edits to export starting from offset ({soffs})'
+            raise s_exc.BadArg(mesg=mesg)
+
+        finished = False
+
+        genr = layer.syncNodeEdits2(soffs, wait=False)
+
+        nodeiter = aiter(genr)
+
+        while not finished:
+
+            try:
+                # Pull the first edit so we can get the starting offset
+                first = await anext(nodeiter)
+            except StopAsyncIteration: # pragma: no cover
+                break
+
+            soffs = first[0]
+
+            with tmpfile(opts.outdir) as (fd, tmppath):
+
+                # Write header to file
+                fd.write(s_msgpack.en((
+                    'init',
+                    {
+                        'hdrvers': 1,
+                        'celliden': celliden,
+                        'cellvers': cellvers,
+                        'layriden': opts.iden,
+                        'offset': soffs,
+                        'chunksize': opts.chunksize,
+                        'tick': s_common.now(),
+                    }
+                )))
+
+                # Now write the first edit that we already pulled
+                fd.write(s_msgpack.en(('edit', first)))
+
+                count = 1
+
+                async for nodeedit in nodeiter:
+
+                    # Write individual edits to file
+                    fd.write(s_msgpack.en(('edit', nodeedit)))
+
+                    eoffs = nodeedit[0]
+
+                    count += 1
+
+                    if opts.chunksize and count % opts.chunksize == 0:
+                        break
+
+                else:
+                    finished = True
+
+                # Write footer to file
+                fd.write(s_msgpack.en(('fini', {
+                    'offset': eoffs,
+                    'tock': s_common.now(),
+                })))
+
+                path = s_common.genpath(opts.outdir, f'{celliden}.{opts.iden}.{soffs}-{eoffs}.nodeedits')
+                os.rename(tmppath, path)
+                outp.printf(f'Wrote layer node edits {soffs}-{eoffs} to {path}.')
+
+            # Save state file after each export file
+            state['offset:next'] = eoffs + 1
+            s_common.yamlsave(state, statefile)
+
+    return 0
+
+async def main(argv, outp=s_output.stdout):
+
+    pars = argparse.ArgumentParser(prog='layer.dump', description=descr)
+    pars.add_argument('--url', default='cell:///vertex/storage',
+                      help='The telepath URL of the Synapse service.')
+    pars.add_argument('--offset', default=None, type=int,
+                      help='The starting offset of the node edits to export.')
+    pars.add_argument('--chunksize', default=0, type=int,
+                      help='The number of node edits to store in a single file. Zero to disable chunking.')
+    pars.add_argument('--outdir', default='.',
```

Comment:
do we want to make this a mandatory argument?

Reply:
I don't think it's necessary. Using the current directory as the outdir default feels right.

Reply:
I think in the long term we are probably better off requiring the user to provide positive control of where their output goes too ( similar to #4333 ). Especially if this is run in a container and its possible that your local directory may be an ephemeral filesystem and your real target location may be a mapped volume.

Reply:
outdir should be after the layer iden

Reply:
I'll make a vertical-hand call. :) Lets go with `<layer> <outdir>` as mandatory positional arguments. :+1: